### PR TITLE
[SYNPY-974] Update descriptions for parameters of generateManifest and fix typo in _generate_folder_manifest

### PR DESCRIPTION
**Pull request by danlu1**

### CODE REVIEW

File: synapseutils/sync.py [hunk 1]
```diff
@@ -614,8 +614,8 @@ def _upload_item(
 
 def generateManifest(syn, allFiles, filename, provenance_cache=None):
     """Generates a manifest file based on a list of entities objects.
-
-    :param allFiles:   A list of File Entities
+    :param syn:   A synapse object as obtained with syn = synapseclient.login()
+    :param allFiles:   A list of File Entities on Synapse
```
Comment (thomasyu888) [inline synapseutils/sync.py:618]:
```suggestion
    :param allFiles:   A list of File Entities on Synapse
```

Comment (thomasyu888) [reply synapseutils/sync.py:618]: Thanks Dan,  does this have to be a list of `File` entity objects, or can it be synapse ids?

File: synapseutils/sync.py [hunk 2]
```diff
@@ -614,8 +614,8 @@ def _upload_item(
 
 def generateManifest(syn, allFiles, filename, provenance_cache=None):
     """Generates a manifest file based on a list of entities objects.
-
-    :param allFiles:   A list of File Entities
+    :param syn:   A synapse object as obtained with syn = synapseclient.login()
+    :param allFiles:   A list of File Entity objects on Synapse (can't be "Synapse ids)
```
Comment (danlu1) [inline synapseutils/sync.py:618]: @thomasyu888 Updated the description of 'allFiles'.

File: synapseutils/sync.py [hunk 3]
```diff
@@ -614,8 +614,8 @@ def _upload_item(
 
 def generateManifest(syn, allFiles, filename, provenance_cache=None):
     """Generates a manifest file based on a list of entities objects.
-
-    :param allFiles:   A list of File Entities
+    :param syn:   A synapse object as obtained with syn = synapseclient.login()
+    :param allFiles:   A list of File Entity objects on Synapse (can't be Synapse ids)
```
Comment (danlu1) [inline synapseutils/sync.py:618]: @thomasyu888 Updated the description of 'allFiles'.

File: synapseutils/sync.py [hunk 4]
```diff
@@ -614,8 +614,8 @@ def _upload_item(
 
 def generateManifest(syn, allFiles, filename, provenance_cache=None):
     """Generates a manifest file based on a list of entities objects.
-
-    :param allFiles:   A list of File Entities
+    :param syn:   A synapse object as obtained with syn = synapseclient.login()
+    :param allFiles:   A list of File Entity objects on Synapse (can't be SynapseIDs)
```
Comment (danlu1) [inline synapseutils/sync.py:618]: @thomasyu888 updated the 'allFiles' description.

Comment (thomasyu888) [inline synapseutils/sync.py:618]:
```suggestion
    :param allFiles:   A list of File Entity objects on Synapse (can't be Synapse IDs)
```

Comment (thomasyu888) [inline synapseutils/sync.py]:
```suggestion
    :param syn: A synapse object as obtained with syn = synapseclient.login()
    :param allFiles: A list of File Entity objects on Synapse (can't be SynapseIDs)
```

**Pull request merged**